client/blocks/comments/post-comment-list.jsx

```diff
@@ -276,7 +277,14 @@ class PostCommentList extends React.Component {
 		const commentText = this.state.commentText;
 
 		// Are we displaying the comment form at the top-level?
-		if ( this.props.activeReplyCommentId && ! this.state.errors ) {
+		if (
+			this.props.activeReplyCommentId ||
+			size(
+				filter( this.props.commentsTree, comment => {
```

Comment:
heres some improvements I think:

1. instead of looping over commentsTree, loop over commentsTree.children.  that ensures they all have parents (although only gives you the IDs)
2. instead of `size( filter (` you can do an `every` check.

Reply:
It'll become something like (pseudocode):
```js
if (
  activeReplyCommentId ||
  every( commentsTree.children, commentId => {
    return get( commentsTree, [ commentId, 'data', 'isPlaceholder' ] );
  } )
```

Reply:
I _think_ that'll only work if all the children are placeholders?

`every` "returns true if all elements pass the predicate check, else false."

https://lodash.com/docs/4.17.4#every

Reply:
likely want `some` instead of `every`.

Reply:
Unfortunately it doesn't look like we can use `commentsTree.children`, because placeholder comments don't appear to be added to it:

<img width="502" alt="screen shot 2017-10-11 at 12 14 35" src="https://user-images.githubusercontent.com/17325/31437681-011b1856-ae7e-11e7-82da-b9fb1b806e42.png">

Reply:
welp. :(. current method it is!
Thank you for trying it!